presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/AddIntermediateAggregations.java

```diff
@@ -177,7 +177,9 @@ private PlanNode addGatheringIntermediate(AggregationNode aggregation, PlanNodeI
                     new AggregationNode.Aggregation(
                             new FunctionCall(QualifiedName.of(aggregation.getSignature().getName()), ImmutableList.of(output.toSymbolReference())),
                             aggregation.getSignature(),
-                            Optional.empty())); // No mask for INTERMEDIATE
+                            Optional.empty(),  // No mask for INTERMEDIATE
+                            entry.getValue().getOrderBy(),
```

Comment:
It can be optimized out. But that's a different thing. Will do that once the execution is settled. When I initially talked with @martint, we agreed that we will implement the functionality first and add optimization later.